### PR TITLE
Add all artifacts to GCE job

### LINKED ISSUE #360: Ensure GCE Jobs Generate Appropriate Artifacts
Today, we have artifact generation and fetching tasks for all the jobs that run off of AWS, as it only requires one hop from the Jenkins master using the `origin-ci-tool` SSH configuration onto the AWS host. We do not have the same level of artifact gathering for the GCE job, as this job uses the AWS host as a intermediary and connects to GCE from there -- so we grab _e.g._ the `origin-node` log from the AWS host, but that host is not running `origin`.

We need to determine if we can get the normal artifact gathering logic to apply cleanly to the GCE jobs (maybe by using SSH with `-J` jump) or if we just need to hard-code a list of artifacts to get us to a better state quickly, if we think the prior will be hard to do and we do not expect to have more jobs on GCE in the future.

/cc @deads2k @smarterclayton

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

Fixes #360

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
test job running: https://ci.openshift.redhat.com/jenkins/job/test_branch_origin_extended_conformance_gce/787/console

**Comment by stevekuznetsov** [comment]
@smarterclayton can you PTAL second comit

File: sjb/config/test_data/gce-artifacts.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+set -o xtrace
+
+base_artifact_dir="/tmp/artifacts"
+
+for instance in $( gcloud compute instances list --regexp ".*${INSTANCE_PREFIX}.*" --uri ); do
+    info="$( mktemp )"
+    gcloud compute instances describe "${instance}" > "${info}"
+    name="$( jq '.name' --raw-output "${info}" | tail -c 5 )"
+    if jq '.tags.items | contains("ocp-master")' --exit-status; then
```
Comment (smarterclayton) [inline sjb/config/test_data/gce-artifacts.sh]: the "ocp-" prefix is configurable but we are unlikely to change it in the near term.  But we might in the future if we do an origin specific one.

**Comment by smarterclayton** [comment]
LGTM if you've tested it.

File: sjb/config/test_data/gce-artifacts.sh (hunk 1 above)
Comment (stevekuznetsov) [inline sjb/config/test_data/gce-artifacts.sh]: Aren't we doing an Origin deployment on these anyway, not `atomic-openshift` ?

**Comment by stevekuznetsov** [comment]
Test job: https://ci.openshift.redhat.com/jenkins/job/test_branch_origin_extended_conformance_gce/790/console

**Comment by stevekuznetsov** [comment]
Moved everything to the Jenkins master .. https://ci.openshift.redhat.com/jenkins/job/test_branch_origin_extended_conformance_gce/791/console

**Comment by stevekuznetsov** [comment]
We don't have `openshift-ansible` installed on those nodes ... new test: https://ci.openshift.redhat.com/jenkins/job/test_branch_origin_extended_conformance_gce/804/console

**Comment by stevekuznetsov** [comment]
Success: https://ci.openshift.redhat.com/jenkins/job/test_branch_origin_extended_conformance_gce/812/console

**Pull request merged**